### PR TITLE
Issue #17165: Remove exceptionVariableName property of EmptyCatchBlock

### PR DESCRIPTION
**Pull request by mohitsatr**
part of : #17563 

Diff Regression config: https://gist.githubusercontent.com/mohitsatr/e1438c3d37ea33baff8301aec7ec41b9/raw/49a29c43657baf6defd598ab9f78263345aaa346/master_emptycatchblock.xml
Diff Regression patch config: https://gist.githubusercontent.com/mohitsatr/5e2fed73f37698ca3ee316e5f42df083/raw/ce74c66e85578bb1628e918a9777eca7edf12135/patch_emptycatchblock.xml

### CODE REVIEW

File: src/it/resources-noncompilable/com/google/checkstyle/test/chapter4formatting/rule4841indentation/InputCatchParametersOnNewLine.java [hunk 1]
```diff
@@ -10,7 +10,7 @@ public void test1() {
     try {
       System.out.println("try0");
     } catch (NullPointerException | IllegalArgumentException expected) {
-    }
+    } // violation above 'Empty catch block.'
```
Comment (Zopsss) [inline src/it/resources-noncompilable/com/google/checkstyle/test/chapter4formatting/rule4841indentation/InputCatchParametersOnNewLine.java]: removing `exceptionVariableName` property causes false-negatives. This exception should have a comment explaining why it was ignored. Please add `// ok until` comments to them.

Comment (mohitsatr) [reply src/it/resources-noncompilable/com/google/checkstyle/test/chapter4formatting/rule4841indentation/InputCatchParametersOnNewLine.java]: done

**Comment by mohitsatr** [comment]
Github, generate report

**Comment by romani** [comment]
diff report clearly shows us, that "expected" is well spread pattern in test code.
and there is no empty catch with comment where we can look on what people write as comment.

**Comment by romani** [comment]
@mohitsatr , please make CI green.

**Comment by romani** [comment]
@mohitsatr , Ideally we need one more diff report, that will show what user write in comment as excuse , we could do this by placing `commentFormat="^$"`, but even it shows what users usually do, google style does not give any recomendation on wording. So any text is ok, default of commentFormat is any.

Right now any comment is excuse for no code in catch.

@mohitsatr , please update this PR to have no "until". All Inputs should have comment in catch or have violation on it, what ever is easier.
Please make sure we have samples with comments in catch.
This PR will fix an issue.

**Comment by Zopsss** [comment]
> what user write in comment as excuse , we could do this by placing commentFormat="^$"

I think we should use `\w+`, it ensures atleast one character is present in the comment, i.e the comment is not empty.

**Comment by romani** [comment]
@mohitsatr , please add test code with empty comment (empty  or some whitesapces) to make sure it is violation

**Comment by mohitsatr** [comment]
Github, generate report

**Comment by romani** [comment]
@mohitsatr , please update commit message to reference right issue

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchEmptyComment.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    }
+  }
+
+  private void foo3() { // comment
+    try {
+      throw new IOException();
+      // violation below 'Empty catch block'
+    } catch (IOException | NullPointerException | ArithmeticException e) {
+      /***/
+    } // violation above 'Javadoc comment is placed in the wrong location'
```
Comment (romani) [inline src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchEmptyComment.java]: `/* */`

make it like.
this is comment, but you places javadoc comment. Attention that you have extra `*` that triggers javdoc checks

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchEmptyComment.java
Comment (Zopsss) [inline src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchEmptyComment.java:1]: please add following cases:


```java
   try {
      throw new RuntimeException();
      // violation below 'Empty catch block'
    } catch (Exception expected) {
      /* */
    }

   try {
      throw new RuntimeException();
    } catch (Exception expected) {
      /* no violation. */
    }

   try {
      throw new IOException();
    } catch (IOException | NullPointerException | ArithmeticException e) { /* no violation. */
    }
```

if you can think of more cases, please add them too

**Review by Zopsss — CHANGES_REQUESTED** [review]

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchEmptyComment.java
Comment (mohitsatr) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchEmptyComment.java:1]: done

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchEmptyComment.java (hunk 2 above)
Comment (mohitsatr) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchEmptyComment.java]: done

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchEmptyComment.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  }
+
+  private void foo4() {
+    try {
+      throw new IOException();
+      // violation below 'Empty catch block'
+    } catch (IOException | NullPointerException | ArithmeticException e) {
+      /* */
+    }
+  }
```
Comment (Zopsss) [inline src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchEmptyComment.java:40]:
```java
  private void some2() { 
     try { 
       throw new IOException(); 
       // violation below 'Empty catch block' 
     } catch (IOException e) { 
       /* */ 
     } 
   }
```

Both of these are identical only number of exceptions are different

**Comment by romani** [comment]
Conflict, please rebase

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchEmptyComment.java (hunk 3 above)
Comment (mohitsatr) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchEmptyComment.java:40]: done

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchBlockViolationsByVariableName.java [hunk 4]
```diff
@@ -23,7 +23,7 @@ private void foo2() {
     try {
       throw new IOException();
     } catch (IOException | NullPointerException | ArithmeticException expected) {
-    }
+    } // violation above 'Empty catch block.'
   }
```
Comment (Zopsss) [inline src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchBlockViolationsByVariableName.java]: why we kept the violation? @mohitsatr 

In other places, comments are added to not throw violation.

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchBlockViolationsByVariableName.java
Comment (Zopsss) [inline src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchBlockViolationsByVariableName.java:1]: I don't think we need this file anymore as Style Guide no longer mentions anything about exception variable name. @mohitsatr @romani wdyt?

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchEmptyComment.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+  private void foo1() {
+    try {
+      throw new RuntimeException();
+      // violation below 'Empty catch block'
+    } catch (Exception e) {
+      //
+    }
+  }
```
Comment (Zopsss) [inline src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchEmptyComment.java]: identical cases

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchEmptyComment.java [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    } catch (IOException | NullPointerException | ArithmeticException e) { //
+    } // violation above 'Empty catch block'
+  }
+
+  private void foo6() {
+    try {
+      throw new IOException();
+    } catch (IOException e) { //
+    } // violation above 'Empty catch block'
+  }
```
Comment (Zopsss) [inline src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchEmptyComment.java]: identical cases. But let's keep it, we should have atleast one case with multiple type of exceptions.

**Review by Zopsss — CHANGES_REQUESTED** [review]
Items:

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchEmptyComment.java
Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchEmptyComment.java:1]: I dont see this case being added:

```java
   try {
      throw new IOException();
    } catch (IOException | NullPointerException | ArithmeticException e) { /* no violation. */
    }
```

please also add following cases:

```java
   try {
      throw new IOException();
    // violation below 'Empty catch block'
    } catch (IOException | NullPointerException | ArithmeticException e) { /* */
    }

   try {
      throw new IOException();
    // violation below 'Empty catch block'
    } catch (IOException | NullPointerException | ArithmeticException e) { /**/
    }
```

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchBlockViolationsByVariableName.java (hunk 4 above)
Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchBlockViolationsByVariableName.java]: I think it is fine to have violaiton, as we migrating from one mode of Check to another, so this violation clearly shows that old "expected" (variable name) is not good any more.

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchBlockViolationsByVariableName.java
Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchBlockViolationsByVariableName.java:1]: lets keep it, it use to be for 10 years, it is time where people already got lifelong habit, and better to show that old norm is not ok any more.

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule412nonemptyblocks/InputFormattedTryCatchIfElse.java [hunk 7]
```diff
@@ -35,18 +35,22 @@ void foo() throws Exception {
 
     try (MyResource r = new MyResource()) {
     } catch (Exception expected) {
+      // ignore
```
Comment (romani) [inline src/it/resources/com/google/checkstyle/test/chapter4formatting/rule412nonemptyblocks/InputFormattedTryCatchIfElse.java]: we need to keep formatted and not-formatted in sync by content

Image

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule412nonemptyblocks/InputFormattedTryCatchIfElse.java (hunk 7 above)
Comment (mohitsatr) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule412nonemptyblocks/InputFormattedTryCatchIfElse.java]: done

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchEmptyComment.java (hunk 6 above)
Comment (mohitsatr) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchEmptyComment.java]: :+1:

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchEmptyComment.java (hunk 5 above)
Comment (mohitsatr) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchEmptyComment.java]: removed

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchEmptyComment.java
Comment (mohitsatr) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchEmptyComment.java:1]: oh my bad. did not create "InputFormatted..." file, so formatter overrode changes. fixed

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputFormattedEmptyCatchEmptyComment.java [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+      throw new IOException();
+    } catch (IOException | NullPointerException | ArithmeticException e) { //
+    } // violation above 'Empty catch block'
+  }
+
+  private void foo6() {
+    try {
+      throw new IOException();
+    } catch (IOException e) { //
+    } // violation above 'Empty catch block'
```
Comment (romani) [inline src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputFormattedEmptyCatchEmptyComment.java]: please put all `// violation ....` under throw.
It is very clear pattern to read such inputs, I like how you did this in other cases below. Please apply to all new snippets.

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchEmptyComment.java [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+      throw new IOException();
+    // violation below 'Empty catch block'
+    } catch (IOException | NullPointerException | ArithmeticException e) { /* */
+    }
+  }
+
+  private void foo11() {
+    try {
+      throw new IOException();
+    // violation below 'Empty catch block'
```
Comment (romani) [inline src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchEmptyComment.java]: please fix indentation of comment here in above block

**Review by romani — CHANGES_REQUESTED** [review]
last minors:

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputFormattedEmptyCatchEmptyComment.java (hunk 8 above)
Comment (mohitsatr) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputFormattedEmptyCatchEmptyComment.java]: done

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchEmptyComment.java (hunk 9 above)
Comment (mohitsatr) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchEmptyComment.java]: done

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchEmptyComment.java (hunk 6 above)
Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchEmptyComment.java]: resolved.

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchEmptyComment.java (hunk 5 above)
Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchEmptyComment.java]: resolved

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchBlockViolationsByVariableName.java
Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchBlockViolationsByVariableName.java:1]: resolved

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchBlockViolationsByVariableName.java (hunk 4 above)
Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchBlockViolationsByVariableName.java]: resolved

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchEmptyComment.java (hunk 3 above)
Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchEmptyComment.java:40]: resolved

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchEmptyComment.java
Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/InputEmptyCatchEmptyComment.java:1]: resolved
